cachito/web/static/api_v1.yaml

```diff
@@ -1010,19 +1073,32 @@ components:
             example: some_flag
         packages:
           type: object
-          items:
-            type: array
-            items:
-              type: object
-          example:
+          properties:
+            gomod:
+              type: array
+              items:
+                $ref: "#/components/schemas/PackageConfig"
             npm:
-            - path: client
-            - path: legacy/client
+              type: array
+              items:
+                $ref: "#/components/schemas/PackageConfig"
+            pip:
+              type: array
+              items:
+                $ref: "#/components/schemas/PipPackageConfig"
+            rubygems:
+              type: array
+              items:
+                $ref: "#/components/schemas/PackageConfig"
+            yarn:
+              type: array
+              items:
+                $ref: "#/components/schemas/PackageConfig"
+          additionalProperties: false
         pkg_managers:
           type: array
           items:
-            type: string
-            example: gomod
+            $ref: "#/components/schemas/PackageManager"
```

Comment:
PackageManager allows `null` right? I think the pkg_managers array shouldn't allow that

https://swagger.io/docs/specification/using-ref/ `$ref and Sibling Elements` makes me sad, otherwise we could make PackageManager not nullable and allow null only when needed :disappointed: 

Maybe something like `allOf` with `- nullable: false` would still work?

Reply:
Good point. I wonder now though if we need the null value at all here and if we're correctly using "default: null" (and the "nullable: true" I added to accommodate it) in other areas of the spec?

It appears that an empty array is still valid for pkg_managers in RequestNew even if nullable is removed from PackageManager.

Reply:
Hmm, yeah that's a good point. I recall thinking it made sense in some context, let me see if I can remember what context that was :smile:

Reply:
Oh yeah, now I remember. I was about a query like

`/requests?per_page=10`

Are the `state` and `pkg_manager` params to this query considered null, or not present? I would hope the latter, in which case not making them nullable sounds like the way to go

Reply:
I believe it's the latter:
- https://swagger.io/docs/specification/describing-parameters/#required-optional
- https://swagger.io/docs/specification/describing-parameters/#nullable

I'm running a few tests now, but I think it's safe to not have them nullable. I'll remove a few other unnecessary ones in a separate commit